### DISCUSSION TITLE
Libraries needed for Metal/Vulkan on macOS?

### DISCUSSION THREAD

**Discussion by jwwalker** [google/filament#3359] [Q&A] [answered]
I set up an Xcode project to build the hello-triangle sample on macOS.  Then I changed its configuration to use Metal rather than OpenGL as the backend.  I expected it wouldn't work without further changes because I was not yet including or linking to libMoltenVK.dylib and libvulkan.1.dylib.  Yet it did work.  Why?  Does libsdl2.dylib or some static library serve the same purpose?

**Comment by romainguy** [comment, marked as answer]
MoltenVK and libvulkan are for Vulkan only, not for Metal. They're not needed unless you absolutely want to use the Vulkan backend on macos (it's only really useful to us since MoltenVK translates Vulkan to Metal).

**Reply by jwwalker** [reply]
OK, that helps!  One thing, though... if I don't need Vulkan, then why do I need the bluevk static library to avoid link errors?

**Reply by romainguy** [reply]
Because you are using a version of filament compiled with Vulkan support. We turned that off in more recent versions I believe.

**Reply by jwwalker** [reply]
Just to be sure, I did a fresh clone of the filament repo and a fresh build with build.sh.  I still get link errors without bluevk.

**Reply by romainguy** [reply]
You can use `-v` with `build.sh` to remove Vulkan entirely.

**Reply by jwwalker** [reply]
I tried deleting the `out` directory and then using `./bulld.sh -i -j -v debug release`, but I'm still getting a link error without `libbluevk.a`.

**Reply by prideout** [reply]
`-v` excludes Vulkan from the Android build, but the easy build script always enables Vulkan for Desktop builds.  If this is problematic we can change it.
